regions/shapes/range.py

```diff
@@ -0,0 +1,72 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from astropy.coordinates import Angle, SkyCoord
+from regions.core.attributes import QuantityLength
+from ..core import SkyRegion, PixelRegion, PixCoord
+
+__all__ = ["RangePixelRegion", "RangeSphericalRegion"]
+
+inf = Angle(float("inf"), "deg")
+
+
+# TODO: needed?
+class RangePixelRegion(PixelRegion):
+    """
+    A range in pixel coordinates.
+
+    See `RangeSphericalRegion`.
+    """
+
+
+class RangeSphericalRegion(SkyRegion):
+    """
+    A range in longitude and latitude.
+
+    Use case: http://www.ivoa.net/documents/SIA/20151223/REC-SIA-2.0-20151223.html#toc12
+
+    Parameters
+    ----------
+    lon_min, lon_max : `~astropy.coordinates.Angle`
+        Range in longitude
+    lat_min, lat_max : `~astropy.coordinates.Angle`
+        Range in latitude
+    frame : `~astropy.coordinates.BaseCoordinateFrame`
+        Sky frame
+    meta : `~regions.RegionMeta` object, optional
+        A dictionary which stores the meta attributes of this region.
+    visual : `~regions.RegionVisual` object, optional
+        A dictionary which stores the visual meta attributes of this region.
+    """
+    # TODO: should we use these properties, or SkyCoord for min / max,
+    # or a center-based representation?
```

Comment:
Please don't use a center-based representation.  That way lies the madness of ADQL's ill-fated BOX.

A true range-based representation is fundamentally not relocatable in latitude - its shape is inherently different for different latitudes.

A geometrically natural center-based representation needs to be fully relocatable - it should have the same shape anywhere on the sphere as long as its size parameters stay the same.  (It's a coordinate-free geometrical concept.)  

But it's *so* likely to be confused with the coordinate-range region that people have a lot of trouble with it.  BOX is being proposed for withdrawal from ADQL because it caused so much trouble of this kind.

Reply:
OK, I'll leave the current representation then, as four angles and a separate frame attribute.